keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/LaunchConfigurationResolver.kt

```diff
@@ -0,0 +1,84 @@
+package com.netflix.spinnaker.keel.ec2.resolvers
+
+import com.netflix.spinnaker.keel.api.ApiVersion
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.ClusterSpec
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.ec2.SPINNAKER_EC2_API_V1
+import com.netflix.spinnaker.keel.plugin.Resolver
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+
+@Component
+/**
+ * A [Resolver] that looks up configuration for defaults such as the key pair in clouddriver and uses that to fill out
+ * missing information in the [ClusterSpec].
+ */
+class LaunchConfigurationResolver(private val cloudDriverCache: CloudDriverCache) : Resolver<ClusterSpec> {
+  override val apiVersion: ApiVersion = SPINNAKER_EC2_API_V1
+  override val supportedKind: String = "cluster"
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  override fun invoke(resource: Resource<ClusterSpec>): Resource<ClusterSpec> =
+    resource.run {
+      copy(
+        spec = spec.withResolvedKeyPairs(spec.locations.account)
+      )
+    }
+
+  private fun ClusterSpec.withResolvedKeyPairs(account: String): ClusterSpec {
+    val defaultKeyPair = cloudDriverCache.defaultKeyPairForAccount(account)
+    assert(defaultKeyPair != null) { "Default key pair missing for account $account in clouddriver config." }
+
+    var defaultLaunchConfig: ClusterSpec.LaunchConfigurationSpec? = null
+    val overrideLaunchConfigs = mutableMapOf<String, ClusterSpec.LaunchConfigurationSpec>()
+
+    if (defaultKeyPair.contains(REGION_PLACEHOLDER)) {
```

Comment:
Can you rename `REGION_PLACEHOLDER` to `REGION`? I found it a bit confusing..

Reply:
Hmm, but that's what it is, a placeholder for the region...

Reply:
Not sure I follow - we can take it offline :)

Reply:
The definition of that constant is:
```kotlin
    private const val REGION_PLACEHOLDER = "{{region}}"
```